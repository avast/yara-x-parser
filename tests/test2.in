rule test
{
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		$a or
		$b
}
