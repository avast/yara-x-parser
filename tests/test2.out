- SOURCE_FILE
  - RULE
    - "rule" RULE_KW
    - " " WHITESPACE
    - "test" IDENTIFIER
    - "\n" WHITESPACE
    - BLOCK_EXPR
      - "{" L_BRACE
      - "\n\t" WHITESPACE
      - STRINGS
        - "strings" STRINGS_KW
        - ":" COLON
        - "\n\t\t" WHITESPACE
        - VARIABLE_STMT
          - "$a" VARIABLE
          - " " WHITESPACE
          - "=" ASSIGN
          - " " WHITESPACE
          - STRING
            - "\"foo\"" STRING
        - "\n\t\t" WHITESPACE
        - VARIABLE_STMT
          - "$b" VARIABLE
          - " " WHITESPACE
          - "=" ASSIGN
          - " " WHITESPACE
          - STRING
            - "\"bar\"" STRING
      - "\n\t" WHITESPACE
      - CONDITION
        - "condition" CONDITION_KW
        - ":" COLON
        - "\n\t\t" WHITESPACE
        - EXPRESSION_STMT
          - EXPRESSION
            - LITERAL
              - "$a" VARIABLE
            - " " WHITESPACE
            - "or" OR_KW
            - "\n\t\t" WHITESPACE
            - LITERAL
              - "$b" VARIABLE
      - "\n" WHITESPACE
      - "}" R_BRACE
  - "\n" WHITESPACE
